presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -0,0 +1,136 @@
+import com.facebook.presto.metadata.SchemaPropertyManager;
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.metadata.TablePropertyManager;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.security.AccessControlManager;
+import com.facebook.presto.security.AllowAllAccessControl;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.sql.tree.ColumnDefinition;
+import com.facebook.presto.sql.tree.CreateTable;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.transaction.TransactionManager;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.facebook.presto.spi.StandardErrorCode.ALREADY_EXISTS;
+import static com.facebook.presto.spi.session.PropertyMetadata.stringSessionProperty;
+import static com.facebook.presto.testing.TestingSession.createBogusTestingCatalog;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestCreateTableTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private final ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("test-%s"));
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testCreateTableDoesNotThrowAlreadyExistsExceptionWhenIfExists()
+            throws Exception
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        TypeManager typeManager = new TypeRegistry();
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Catalog bogusTestingCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        AtomicReference<Boolean> metadataCreateTableCalled = new AtomicReference<>(Boolean.FALSE);
+
+        MetadataManager metadata = new MetadataManager(
```

Comment:
This is rather messy. The reason `Metadata` is an interface is to allow mocking it for tests. We could create a `MockMetadata`.